components/fxa-client/src/scoped_keys.rs

```diff
@@ -89,6 +103,94 @@ impl ScopedKeysFlow {
         .to_string())
     }
 
+    fn key_pair(self) -> KeyPair<Ephemeral> {
+        self.key_pair
+    }
+
+    fn get_secret_from_ikm(
+        ikm: InputKeyMaterial,
+        apu: &str,
+        apv: &str,
+        alg: &str,
+    ) -> Result<digest::Digest> {
+        let secret = ikm.derive(|z| {
+            let counter = 1;
+            let mut buf: Vec<u8> = vec![];
+            buf.extend_from_slice(&to_32b_buf(counter));
+            buf.extend_from_slice(&z);
+            // otherinfo
+            buf.extend_from_slice(&to_32b_buf(alg.len() as u32));
+            buf.extend_from_slice(alg.as_bytes());
+            buf.extend_from_slice(&to_32b_buf(apu.len() as u32));
+            buf.extend_from_slice(apu.as_bytes());
+            buf.extend_from_slice(&to_32b_buf(apv.len() as u32));
+            buf.extend_from_slice(apv.as_bytes());
+            buf.extend_from_slice(&to_32b_buf(256));
+            digest::digest(&digest::SHA256, &buf)
+        })?;
+        Ok(secret)
+    }
+
+    fn get_peer_public_from_jwk(jwk: &str) -> Result<Vec<u8>> {
+        let jwk_value: serde_json::Value = serde_json::from_str(jwk)?;
+        let x = jwk_value
+            .as_object()
+            .unwrap()
+            .get("x")
+            .unwrap()
+            .as_str()
+            .unwrap();
+        let y = jwk_value
+            .as_object()
+            .unwrap()
+            .get("y")
+            .unwrap()
+            .as_str()
+            .unwrap();
+        let x = base64::decode_config(x, base64::URL_SAFE_NO_PAD)?;
+        let y = base64::decode_config(y, base64::URL_SAFE_NO_PAD)?;
+        let mut peer_public_key: Vec<u8> = vec![0x04];
+        peer_public_key.extend_from_slice(&x);
+        peer_public_key.extend_from_slice(&y);
+        Ok(peer_public_key)
+    }
+
+    pub fn encrypt_keys_jwe(self, jwk: &str, data: &[u8]) -> Result<String> {
```

Comment:
@rfk is it ok to keep this here, even for https://github.com/mozilla/application-services/issues/3122#issuecomment-628965909 ?